qemu/tests/timedrift_check_when_hotplug_vcpu.py

```diff
@@ -0,0 +1,64 @@
+import re
+import time
+import logging
+
+from avocado.utils import process
+
+from virttest import error_context
+
+
+@error_context.context_aware
+def run(test, params, env):
+    """
+    Check time offset after hotplug a vcpu.
+
+    1) sync host time with ntpserver
+    2) boot guest with '-rtc base=utc,clock=host,driftfix=slew'
+    3) stop auto sync service in guest (rhel7 only)
+    4) sync guest system time with ntpserver
+    5) hotplug a vcpu by qmp command
+    6) query guest time offset with ntpserver for several times
+
+    :param test: QEMU test object.
+    :param params: Dictionary with test parameters.
+    :param env: Dictionary with the test environment.
+    """
+    clock_sync_command = params["clock_sync_command"]
+
+    error_context.context("Sync host system time with ntpserver", logging.info)
+    process.system(clock_sync_command, shell=True)
+
+    vm = env.get_vm(params["main_vm"])
+    session = vm.wait_for_login()
+
+    ntp_query_cmd = params.get("ntp_query_cmd", "")
+    query_times = int(params.get("query_times", "4"))
+    query_internal = float(params.get("query_internal", "600"))
+    drift_threshold = float(params.get("drift_threshold", "3"))
+
+    error_context.context("Stop auto sync service in guest", logging.info)
+    cmd = "service chronyd stop"
+    session.cmd_status_output(cmd)
+
+    error_context.context("Sync time from guest to ntpserver", logging.info)
+    session.cmd(clock_sync_command)
+
+    error_context.context("Hotplug a vcpu to guest", logging.info)
+    if int(params["smp"]) <= int(params["vcpus_maxcpus"]):
+        vm.hotplug_vcpu()
+        time.sleep(1)
+    else:
+        test.error("Invalid operation, valid index range 0:%d, used range 0:%d"
+                   % (int(params["vcpus_maxcpus"])-1, int(params["smp"]) - 1))
+
+    error_context.context("Check time offset via ntp server", logging.info)
+    for query in range(query_times):
+        output = session.cmd_output(ntp_query_cmd)
+        try:
+            offset = re.findall(r"([+-]\d+\.\d+)", output, re.M)[-1]
+        except IndexError:
+            offset = 0.0
```

Comment:
@xiangchunfu Please help confirm this logic, to zero the offset when not get it in query output. Thanks.

Reply:
@vivianQizhu You are right, It is not reasonable.  If IndexError, it should Raise an error.  Siliu will update it.